addon/components/file-browser/component.js

```diff
@@ -294,9 +272,10 @@ export default Ember.Component.extend({
                         this.get('_items').removeObject(replacedItem);
                         this.notifyPropertyChange('_items');
                     }, 1800);
+                    // Later to avoid flash() trying to set on a destroyed item.
+                    Ember.run.later(() => replacedItem.unloadRecord(), 2200);
```

Comment:
I'm not really happy with this, but...

Reply:
Alternative would be to allow flash() to take a callback?

Reply:
We could do that. I also never had problems with flash being called in a destroyed item, which I guess I should've.